File: ProyectoCivitasRuby/lib/modelo/sorpresa_por_casa_hotel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class SorpresaPorCasaHotel < Sorpresa
    public_class_method :new 
    def initialize(valor, texto)
      super(texto)
      @valor = valor  
    end
  
    #@Override
    def aplicarajugador(actual, todos)
      if(jugador_correcto(actual, todos))
        informe(actual, todos)
        todos[actual].modificar_saldo(@valor*todos[actual].cantidad_casas_hoteles)     
      end
    end
    
    def to_s
      salida = "{@texto}\nSorpresa #{@valor} por casa/hotel"
    end
  end
end

File: ProyectoCivitasJava/src/civitas/SorpresaPorCasaHotel.java
/**
 * Proyecto Civitas
 * Sergio Hervás Cobo
 * Javier Suárez Lloréns
 * Programación y Diseño Orientado a Objetos
 * 2º B3 - Grado en Ingeniería Informática
 * E.T.S.I. Informática y de Telecomunicación
 * Universidad de Granada
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author sergio
 */
public class SorpresaPorCasaHotel extends Sorpresa {
    private String texto;
    private int valor;

    SorpresaPorCasaHotel(int valor, String texto) {
        super(texto);
        this.valor = valor;
    }
    
    @Override
    void aplicarAJugador(int actual, ArrayList <Jugador> todos){
        if(jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            todos.get(actual).modificarSaldo(valor*todos.get(actual).cantidadCasasHoteles());
        }        
    }
    
    @Override
    public String toString(){
        String salida = "Sorpresa " + valor + " por casa/hotel";
        return salida;
    }
    
    
}
